test/functional/currency/currency-parser.js

```diff
@@ -0,0 +1,146 @@
+define([
+	"globalize",
+	"json!cldr-data/main/de/currencies.json",
+	"json!cldr-data/main/de/numbers.json",
+	"json!cldr-data/main/en/currencies.json",
+	"json!cldr-data/main/en/numbers.json",
+	"json!cldr-data/main/zh/currencies.json",
+	"json!cldr-data/main/zh/numbers.json",
+	"json!cldr-data/supplemental/currencyData.json",
+	"json!cldr-data/supplemental/likelySubtags.json",
+	"json!cldr-data/supplemental/plurals.json",
+	"../../util",
+
+	"globalize/currency",
+	"globalize/number",
+	"globalize/plural"
+], function( Globalize, deCurrencies, deNumbers, enCurrencies, enNumbers, zhCurrencies,
+	zhNumbers, currencyData, likelySubtags, plurals, util ) {
+
+var teslaS = 69900;
+
+function extraSetup() {
+	Globalize.load(
+		currencyData,
+		deCurrencies,
+		deNumbers,
+		enCurrencies,
+		enNumbers,
+		plurals,
+		zhCurrencies,
+		zhNumbers
+	);
+
+}
+
+QUnit.module( ".currencyParser( [options] )", {
+	setup: function() {
+		Globalize.load( likelySubtags, {
+			main: {
+				en: {},
+				"en-IN": {},
+				"tr-CY": {}
+			}
+		});
+		Globalize.locale( "en" );
+	},
+	teardown: util.resetCldrContent
+});
+
+QUnit.test( "should validate parameters", function( assert ) {
+	util.assertParameterPresence( assert, "currency", function() {
+		Globalize.currencyParser();
+	});
+
+	util.assertCurrencyParameter( assert, "currency", function( invalidValue ) {
+		return function() {
+			Globalize.currencyParser( invalidValue );
+		};
+	});
+
+	util.assertPlainObjectParameter( assert, "options", function( invalidValue ) {
+		return function() {
+			Globalize.currencyParser( invalidValue );
+		};
+	});
+});
+
+QUnit.test( "should validate CLDR content", function( assert ) {
+	util.assertCldrContent( assert, function() {
+		Globalize.currencyParser();
+	});
+});
+
+QUnit.test( "should return a currency formatter", function( assert ) {
+	var de, zh;
+
+	extraSetup();
+
+	de = Globalize( "de" );
+	zh = Globalize( "zh" );
+
+	assert.equal( Globalize.currencyParser()( "$69,900.00" ), teslaS );
```

Comment:
Parser should know what's the expected currency and assert that. We need a test where a different currency is appropriately not parsed (e.g., parser could return `null`).

Reply:
Yeah I think my parser is not strict enough to pick up `NaN`

Reply:
I mean  this:
```js
assert.equal( Globalize.currencyParser("USD")( "$69,900.00" ), teslaS );
// > 69900

assert.equal( Globalize.currencyParser("EUR")( "$69,900.00" ), teslaS );
// > null
```

I said `null`, but it could be `NaN` instead (like numberParser)